### PR TITLE
Make Electronic an Abstact Class #52

### PR DESCRIPTION
**Pull request by techwulf**
The entities disappear when electronic's view inherits from inanimate, so it still inherits from entity.

### CODE REVIEW

**Comment by Master-Foo** [comment]
The Electronics are going to need to inherit from inanimate, otherwise, there will be bugs in the future.
look at [/source/entities/inanimate/rock/](/source/entities/inanimate/rock/) as an example of an inanimate that works.

One other problem. Don't put the electronic directory inside the inanimate directory. You have it like this: `/source/abstract/inanimate/electronic/`

Put it directly in the abstract directory, like this: `/source/abstract/electronic/`

Take care of those details and I'll probably accept the pull request.

**Comment by jrslagle** [comment]
@Master-Foo
Just so I can keep helping Tom, let me see if I understand.

When you say "Electronics are going to need to inherit from inanimate" do you mean electronic entities should not be inheriting from electronic? Or do you mean "definitely fix that electronic/view inheriting from entity bug"?

The rest makes sense.

**Comment by jrslagle** [comment]
We were also wondering which structure we should clean towards. 

Battery/model.py
Battery/controller.py
Battery/view.py

or 
Battery/model/model.py
Battery/controller/controller.py
Battery/view/view.py 

Both exist.

**Comment by Master-Foo** [comment]
The ancestry of a battery should look like this:
Battery -> Electronic -> Inanimate -> Entity

So, battery is "non-abstract" because it is used directly in the game. So, it goes in the normal place. However, Electronic, Inanimate, and Entity are **ONLY EVER** inherited. So, they go in abstract.

The structure where every part is in it's own folder is better. I've been meaning to get around to clean it up. However, unless you plan on making Battery more complicated (code over one page scroll) don't make it a priority. It becomes a priority when various pieces of logic can be divided into "families"

See the Avatar Controller package for an example on why the controller eventually gets broken up as it becomes more complicated.

**TLDR;** It's my mess, I should clean battery up. However, if you are poking around in there, and feel generous, you can divide it up into folders.

**Comment by jrslagle** [comment]
So we did set up the ancestry just like you said. But for some reason, the entities you convert no longer appear on the hectare - and the game doesn't crash. I narrowed it down to where you can reproduce the bug with this one word. If you go to 
source/abstract/entities/inanimate/electronic line 4

from source.abstract.entities.entity import view

And change entities.entity to entities.inanimate, the entities disappear in game. It also seemed to be the ones with their files split up into separate MCV folders. I tried tinkering with different possibilities, but didn't figure it out, and just left it pointing at .entity for you to look at.

**Comment by Master-Foo** [comment]
Push what you have to your repository and I'll clone it and take a look.

These things are usually pretty simple for such a strange problem. Most likely you forgot to call the ancestor's function somewhere like this:

```python
class Electronic(Inanimate):
    def __init__(self):
        Electoronic.__init__(self)
        #do stuff
        pass

    def some_function(self):
        Electronic.some_function(self):
        #do stuff
        pass
```

Anyway, push and I'll take a look.

**Comment by Master-Foo** [comment]
@techwulf @jrslagle I just pulled and ran @techwulf's latest commit on his electronics branch on his fork. Everything seems to be rendering fine, so I can't really give any advice if I can't reproduce the problem.

In fact, as far as I can tell, the inheritance structure works just fine. The **only** problem is that the abstract electronic directory is in the wrong location. It should be one directory up. But that's just a matter of convention. The code runs perfectly as I can tell.

So... If you guys are having a problem. I can't tell what it is based on what I have to work with...

Just move the electronic directory and change the imports so the electronic entities can find it and it should be a clean pull.
